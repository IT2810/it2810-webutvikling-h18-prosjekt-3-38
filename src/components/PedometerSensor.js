import React from 'react'
import { Pedometer } from 'expo'
import { Text, View } from 'react-native'
import styled from 'styled-components'
import { Asset, Font } from 'expo'

const StyledView = styled.View`
  flex: 1;
  margin-top: 15;
  align-items: center;
  justify-content: center;
`

export default class PedometerSensor extends React.Component {
  state = {
    isPedometerAvailable: 'checking',
    pastStepCount: 0,
    currentStepCount: 0,
    fontLoaded: false

  }

  async componentDidMount () {
    this._subscribe()

    await Font.loadAsync({
      'Roboto-Medium': require('../../assets/fonts/Roboto-Bold.ttf')
    }).then(() => {
      this.setState({ fontLoaded: true })
    })
  }

  componentWillUnmount () {
    this._unsubscribe()
  }

  _subscribe = () => {
    this._subscription = Pedometer.watchStepCount(result => {
      this.setState({
        currentStepCount: result.steps
      })
    })

    Pedometer.isAvailableAsync().then(
      result => {
        this.setState({
          isPedometerAvailable: String(result)
        })
      },
      error => {
        this.setState({
          isPedometerAvailable: 'Could not get isPedometerAvailable: ' + error
        })
      }
    )

    const end = new Date()
    const start = new Date()
    start.setDate(end.getDate() - 1)
    Pedometer.getStepCountAsync(start, end).then(
      result => {
        this.setState({ pastStepCount: result.steps })
      },
      error => {
        this.setState({
          pastStepCount: 'Could not get stepCount: ' + error
        })
      }
    )
  }

  _unsubscribe = () => {
    this._subscription && this._subscription.remove()
    this._subscription = null
  }

  render () {
    return (
      <StyledView>
        <View>
          {this.state.fontLoaded == true ? (<Text style={{ backgroundColor: '#01364c', fontFamily: 'Roboto-Medium', bottom: 20, fontSize: 20, color: 'white' }}>Walk! And watch this go up:</Text>)
            : <Text>Loading...</Text>}
        </View>
        <View>
          {this.state.fontLoaded == true ? (<Text style={{ backgroundColor: '#01364c', fontFamily: 'Roboto-Medium', bottom: 20, fontSize: 20, color: 'white' }}>
            <Text style={{fontSize: 50}}>{ this.state.currentStepCount}</Text>
          </Text>)
            : <Text>Loading...</Text>}
        </View>
      </StyledView>
    )
  }
}
