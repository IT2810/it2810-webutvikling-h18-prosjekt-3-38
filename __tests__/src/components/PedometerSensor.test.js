import React from 'react'
import renderer from 'react-test-renderer'
import PedometerSensor from '../../../src/components/PedometerSensor'

describe('PedometerSensor', () => {
  const component = renderer.create(<PedometerSensor></PedometerSensor>)

  test('renders correctly', () => {
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('has correct initial state', () => {
    expect(component.getInstance().state.isPedometerAvailable).toBe('checking')
  })

  test('updated state correctly', () => {
    component.getInstance().setState(
      {
        pastStepCounter: 3,
        currentStepCounter: 5
      })

    expect(component.getInstance().state.pastStepCounter).toBe(3)
    expect(component.getInstance().state.currentStepCounter).toBe(5)
  })
})
