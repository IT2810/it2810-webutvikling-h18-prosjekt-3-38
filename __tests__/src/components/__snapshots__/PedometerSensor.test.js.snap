// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PedometerSensor renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "marginTop": 15,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Steps taken in the last 24 hours: 
    0
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Walk! And watch this go up: 
    0
  </Text>
</View>
`;
